python/cuml/metrics/_classification.py

```diff
@@ -0,0 +1,106 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import cupy as cp
+import numpy as np
+from cuml.common.memory_utils import with_cupy_rmm
+from cuml.common import input_to_cuml_array
+
+
+@with_cupy_rmm
+def log_loss(y_true, y_pred, eps=1e-15, normalize=True, sample_weight=None):
+    """ Log loss, aka logistic loss or cross-entropy loss.
+    This is the loss function used in (multinomial) logistic regression
+    and extensions of it such as neural networks, defined as the negative
+    log-likelihood of a logistic model that returns ``y_pred`` probabilities
+    for its training data ``y_true``.
+    The log loss is only defined for two or more labels.
+
+    Parameters
+    ----------
+    y_true : array-like, shape = (n_samples,)
+    y_pred : array-like of float,
+        shape = (n_samples, n_classes) or (n_samples,)
+    eps : float
+        Log loss is undefined for p=0 or p=1, so probabilities are
+        clipped to max(eps, min(1 - eps, p)).
+    normalize : bool, optional (default=True)
+        If true, return the mean loss per sample.
+        Otherwise, return the sum of the per-sample losses.
+    sample_weight : array-like of shape (n_samples,), default=None
+        Sample weights.
+
+    Returns
+    -------
+    loss : float
+    Examples
+    --------
+    >>> from cuml.metrics import log_loss
+    >>> import numpy as np
+    >>> log_loss(np.array([1, 0, 0, 1]),
+    ...          np.array([[.1, .9], [.9, .1], [.8, .2], [.35, .65]]))
+    0.21616...
+    References
+    ----------
+    C.M. Bishop (2006). Pattern Recognition and Machine Learning. Springer,
+    p. 209.
+    Notes
+    -----
+    The logarithm used is the natural logarithm (base-e).
+    """
+    y_true, n_rows, n_cols, ytype = \
+        input_to_cuml_array(y_true, check_dtype=[np.int32, np.int64,
+                                                 np.float32, np.float64])
+
+    y_true = y_true.to_output()
+    if y_true.dtype.kind == 'f' and np.any(y_true != y_true.astype(int)):
+        raise ValueError("Continuous format of y_true "
+                         "is not supported by log_loss")
+    if y_true.min() < 0:
+        raise ValueError("The minimum value of y_true "
+                         "cannot be negative")
+
+    y_pred, _, _, _ = \
+        input_to_cuml_array(y_pred, check_dtype=[np.int32, np.int64,
+                                                 np.float32, np.float64],
+                            check_rows=n_rows)
+
+    y_pred = y_pred.to_output()
+    y_true_max = y_true.max()
+    if (y_pred.ndim == 1 and y_true_max > 1) \
+       or (y_pred.ndim > 1 and y_pred.shape[1] <= y_true_max):
+        raise ValueError("The shape of y_pred doesn't "
+                         "match the number of classes")
+
+    y_true = y_true.astype('int32')
```

Comment:
Yes, it is needed. It is used as column indices later on so it has to be int. And `int32` is actually sufficient in practice because `0<=y_true<=number of classes`.